src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala

```diff
@@ -142,6 +147,12 @@ private[actions] abstract class CreateActionBase(dataManager: IndexDataManager)
         resolvedIndexedColumns)
   }
 
+  private def leafFileInfo(path: Path): Seq[FileInfo] = {
+    // Assuming index directories don't contain nested directories. Only Leaf files.
+    val fs = path.getFileSystem(new Configuration)
+    Try { fs.listStatus(path).map(FileInfo(_)).toSeq }.getOrElse(Seq())
```

Comment:
When would this throw an exception and would it be ok to silently ignore?

Reply:
This will throw exception when `path` does not exist. This will happen **before** the index got created. This method is used in the `def logEntry` which is called multiple times during the index creation.
During validation phase for example, the `path` doesn't exist. At this time, we expect an exception when listing index files.

Once index has been created, the `path` exists and we don't expect this exception. I will update this section to identify file not found exception and other exceptions in the next iteration
Thanks @imback82

Reply:
Thanks @imback82 , updated